engine_details/development/debugging/using_cpp_profilers.rst

```diff
@@ -12,150 +12,31 @@ To this end, profilers are useful tools.
     but using C++ profiler may be useful in cases where the GDScript profiler
     is not accurate enough or is missing information due to bugs in the profiler.
 
-Recommended profilers
----------------------
+There are two main types of profilers: sampling profilers and tracing profilers.
 
-- `VerySleepy <http://www.codersnotes.com/sleepy/>`__ (Windows only)
-- `HotSpot <https://github.com/KDAB/hotspot>`__ (Linux only)
-- `Xcode Instruments <https://developer.apple.com/xcode/>`__ (macOS only)
+Sampling profilers periodically interrupt the running program and take a "sample",
+which records which functions are running. The profiler can aggregate this data to
+produce a graph which shows which functions the program spent the most time in.
 
-These profilers may not be the most powerful or flexible options, but their
-standalone operation and limited feature set tends to make them easier to use.
+Tracing profilers work by recording application-specific events over a short
+period of time (usually only a few seconds), producing a log called a "trace".
+The profiler can use the trace to produce a graph showing a high-level timeline of
+what happened, when and how long it took.
 
-Setting up Godot
-----------------
+While a sampling profiler doesn't require any code changes (only a build with
+debug symbols), in order to use a tracing profiler, there needs to be additional
+logging code added.
```

Comment:
Since Godot supports both, and needs to be compiled from scratch for both, I don't think this information is needed.

```suggestion
```